modules/backend/widgets/form/assets/js/october.form.js

```diff
@@ -220,6 +251,157 @@
+
+            if (!field || !field.className) {
+                continue
+            }
+
+            var fieldClasses = field.className.split(/\s+/).map(function (className) {
+                var fieldClass = className.match(/^([a-z\-]+)-field$/)
+                if (fieldClass) {
+                    return fieldClass[1]
+                } else {
+                    return null
+                }
+            }).filter(function (className) {
+                return (className != null)
+            })
+
+            if (fieldClasses.length === 0) {
+                continue
+            }
+
+            // Load validator depending on field type
+            switch (fieldClasses[0]) {
+                case 'text':
+                case 'password':
+                case 'number':
+                case 'textarea':
+                    this._validateText($form, field)
+                    break
+                case 'dropdown':
+                    this._validateDropdown($form, field)
+                    break
+                case 'balloon-selector':
+                    this._validateBalloonSelector($form, field)
+                    break
+            }
+        }
+    }
+
+    FormWidget.prototype.fieldResponseHandler = function ($form, $field, data, status) {
+        if (status !== 'success') {
+            return
+        }
+
+        if (data.valid !== undefined && data.valid === false) {
+            this.showFieldError($field, data.message)
+        }
+    }
+
+    FormWidget.prototype.showFieldError = function ($field, message) {
+        $field.addClass('has-error')
+
+        var $error = $('<div class="validation-error"></div>')
+        $error.text(message)
+
+        $field.append($error)
+    }
+
+
+    FormWidget.prototype.clearFieldError = function ($field) {
+        $field.removeClass('has-error')
+        $field.find('.validation-error').remove()
+    }
+
+    FormWidget.prototype._validateText = function ($form, field) {
+        var $elem = this.$el,
+            $field = $(field),
+            widget = this
+
+        var innerField = field.querySelector('input,textarea')
+        if (!innerField) {
+            return
+        }
+
+        innerField.addEventListener('blur', function (ev) {
+            widget.clearFieldError($field)
+
+            $elem.request('onValidateField', {
+                data: {
+                    fieldId: field.id,
+                    fieldName: field.dataset.fieldName
+                },
+                form: $form,
+                success: function (data, status, jqXHR) {
+                    widget.fieldResponseHandler($form, $field, data, status, jqXHR)
+                }
+            })
+        })
+    }
+
+    FormWidget.prototype._validateDropdown = function ($form, field) {
+        var $elem = this.$el,
+            $field = $(field),
+            widget = this
+
+        var $innerField = $(field.querySelector('select'))
+        if (!$innerField) {
```

Comment:
@bennothommo I think you have a bug here :) `$innerField` will be an empty object, but present.
You could either change it to `!!$innerField` or `$innerField.length === 0`.

Reply:
Good catch :)